src/integrations/HttpIntegration.ts

```diff
@@ -252,7 +253,13 @@ class HttpIntegration implements Integration {
                             res.on('end', () => {
                                 try {
                                     if (chunks && chunks.length) {
-                                        const responseBody: string = Buffer.concat(chunks).toString('utf8');
+                                        const concatedChunks = Buffer.concat(chunks);
+                                        const contentEncoding = HTTPUtils.obtainIncomingMessageEncoding(res);
+                                        const responseBody: string =
```

Comment:
Response body size should also be checked after decoding because decoded response size might be 4x-5x of taken encoded response size.

Reply:
Control added in 'getPayload' method